provider/resources.go

```diff
@@ -7095,7 +7127,19 @@ func Provider() *tfbridge.ProviderInfo {
 
 		// We have ensured that this resource is using upstream's generic tagging
 		// mechanism, so override check so it works.
-		prov.Resources[key].PreCheckCallback = applyTags
+		if callback := prov.Resources[key].PreCheckCallback; callback != nil {
+			prov.Resources[key].PreCheckCallback = func(
+				ctx context.Context, config resource.PropertyMap, meta resource.PropertyMap,
+			) (resource.PropertyMap, error) {
+				config, err := callback(ctx, config, meta)
+				if err != nil {
+					return nil, err
+				}
+				return applyTags(ctx, config, meta)
```

Comment:
I'm not sure I understand the logic here.

- if a resource that hits this code path has a non-nil callback, we want to apply tags _after_ applying the existing callback function to the config?
- if a resource  that hits this code path _doesn't_ have a callback, or it is nil, we behave as before

Since `rds.Instance` is currently the only resource with an explicit callback, could there ever be a new PreCheckCallback that _doesn't_ want this to happen?

Reply:
If there is a callback, we give re-assign `prov.Resources[key].PreCheckCallback` so both the old callback and `applyTags` are called. Otherwise we just set `applyTags`.

If we introduce a new callback where we don't want this to happen, we can change this code. I don't expect that we will though.

Reply:
> If there is a callback, we give re-assign prov.Resources[key].PreCheckCallback so both the old callback and applyTags are called. Otherwise we just set applyTags.

Makes sense!

> If we introduce a new callback where we don't want this to happen, we can change this code. I don't expect that we will though.
How will a future maintainer know to change this code? 
If we don't expect new callbacks to be introduced - should we use the resource name as an explicit trigger here?

Reply:
> How will a future maintainer know to change this code?

If a future maintainer wants to add a new `PreCheckCallback` to a specific resource that overrides the behavior of tags, they will notice their change is not working and investigate. It's all in this file. They can grep for `PreCheckCallback` in `resources.go` and find this.

> If we don't expect new callbacks to be introduced - should we use the resource name as an explicit trigger here?

I'm not sure what you mean? Can you elaborate?

Reply:
> they will notice their change is not working and investigate.
I don't think that is ideal. I think the design of PreCheckCallback invites one to think "oh neat, I can add a callback in this place, on a resource". Getting a surprise can be really frustrating, regardless of how easy it would be to grep for PreCheckCallback.

> Can you elaborate?
If this particular callback is unique to `rds.Instance`, I think I would prefer logic that explicitly triggers on "if this resource is `rds.Instance`, use this callback". That will avoid a future maintainer having to look why something seemingly unrelated is getting involved with their callback.

I guess my TL;DR is: I don't think that "PreCheckCallback is not nil" is a discerning enough evaluation to decide whether we should call this code. Let's just use `rds.Instance`. After all, we would want to _remove_ this on v7, correct?